src/base/settingvalue.h

```diff
@@ -40,8 +40,15 @@ class CachedSettingValue
     using ProxyFunc = std::function<T (const T&)>;
 
 public:
-    explicit CachedSettingValue(const char *keyName, const T &defaultValue = T()
-            , ProxyFunc proxyFunc = [](const T &value) { return value; })
```

Comment:
>I was sure the compiler optimizes it

I just want to make sure `` isn't involved when I don't intend to.
And `std::function` do type erasures anyways, I don't think optimizers can do much about it.